core/src/main/java/org/geysermc/geyser/translator/inventory/chest/DoubleChestInventoryTranslator.java

```diff
@@ -76,11 +77,25 @@ public void prepareInventory(GeyserSession session, Inventory inventory) {
                 dataPacket.setData(tag.build());
                 dataPacket.setBlockPosition(session.getLastInteractionBlockPosition());
                 session.sendUpstreamPacket(dataPacket);
-                return;
+
+                return true;
+            }
+        }
+
+        // Check if a fake block can be placed, either above the player or beneath.
+        BedrockDimension dimension = session.getChunkCache().getBedrockDimension();
+        int minY = dimension.minY(), maxY = minY + dimension.height();
+        Vector3i position = session.getPlayerEntity().getPosition().toInt().add(0, 5, 0);
```

Comment:
This code can probably be optimized a bit to reuse variables since `toInt()` might be called twice, which means two allocations.

Also IntelliJ is complaining about duplicate code fragment around here but I'm not sure if there's an easy way to centralize the code.